web/packages/teleterm/src/ui/ConnectMyComputer/NavigationMenu.tsx

```diff
@@ -21,39 +21,62 @@ import { Box } from 'design';
 
 import { useAppContext } from 'teleterm/ui/appContextProvider';
 import { ListItem } from 'teleterm/ui/components/ListItem';
+import { ClusterUri } from 'teleterm/ui/uri';
 
-import { ConnectMyComputerIcon } from './ConnectMyComputerIcon';
+import { NavigationMenuIcon } from './NavigationMenuIcon';
+import { canUseConnectMyComputer } from './permissions';
 
-export function ConnectMyComputer() {
+interface NavigationMenuProps {
+  clusterUri: ClusterUri;
+}
+
+export function NavigationMenu(props: NavigationMenuProps) {
   const iconRef = useRef();
   const [isPopoverOpened, setIsPopoverOpened] = useState(false);
   const appCtx = useAppContext();
+  const cluster = appCtx.clustersService.findCluster(props.clusterUri);
```

Comment:
I'd add a comment above it that `DocumentCluster` renders this component only if the cluster exists.